### PR TITLE
Max's PR for Tor code

### PR DESCRIPTION
**Pull request by oconnor663**
@maxtaco

### CODE REVIEW

File: go/libkb/proof_support_web.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ func NewWebChecker(p RemoteProofChainLink) (*WebChecker, ProofError) {
 	return &WebChecker{p}, nil
 }
 
+func (rc *WebChecker) GetTorError() ProofError {
+	urlBase := rc.proof.ToDisplayString()
+	if !strings.HasPrefix(urlBase, "https:") {
```
Comment (oconnor663) [inline go/libkb/proof_support_web.go]: Worth using `url.Parse` for this?

File: go/libkb/id_table.go [hunk 2]
```diff
@@ -1206,9 +1218,6 @@ func (idt *IdentityTable) proofRemoteCheck(hasPreviousTrack, forceRemoteCheck bo
 		}
 	}
 
-	var pc ProofChecker
-	pc, res.err = NewProofChecker(p)
-
 	if res.err != nil {
```
Comment (oconnor663) [inline go/libkb/id_table.go]: I think we need to move this check up. We don't want to clobber a real proof error with a Tor warning, right?

File: go/libkb/api.go [hunk 3]
```diff
@@ -316,12 +316,12 @@ func (a *InternalAPIEngine) consumeHeaders(resp *http.Response) error {
 func (a *InternalAPIEngine) fixHeaders(arg APIArg, req *http.Request) {
```
Comment (oconnor663) [inline go/libkb/api.go]: Do we also want to strip out `User-Agent` and `X-Keybase-Client` headers? Should we just have a policy of "Tor mode = no headers"?

Comment (oconnor663) [inline go/libkb/api.go]: Or rather, "strict tor mode = ..."

File: go/libcmdline/cmdline.go [hunk 4]
```diff
@@ -352,6 +367,18 @@ func (p *CommandLine) PopulateApp(addHelp bool, extraFlags []cli.Flag) {
 			Name:  "api-timeout",
 			Usage: "set the HTTP timeout for API calls to the keybase API server",
 		},
+		cli.StringFlag{
+			Name:  "tor-mode",
+			Usage: "set TOR mode to be 'leaky', 'none', or 'strict'. 'none' by default",
```
Comment (oconnor663) [inline go/libcmdline/cmdline.go]: Is there space to document that "strict tries to be anonymous even to the Keybase server, leaky uses Tor for your connection but identifies you to the server"?

Comment (maxtaco) [inline go/libcmdline/cmdline.go]: we can do something like `keybase help tor` ?

Comment (oconnor663) [inline go/libcmdline/cmdline.go]: I'm fine with anything, as long as there's some way for the user reading this help string to figure out how to get to the definition of these modes.

**Comment by oconnor663** [comment]
Otherwise: :shipit: !!!

**Comment by oconnor663** [comment]
landed

**Pull request closed without merging**